mkat_fpga_tests/test_cbf.py

```diff
@@ -336,6 +339,124 @@ def print_baselines():
             fengops.feng_eq_set(self.correlator, source_name=inp, new_eq=old_eqs)
             zero_inputs.remove(inp)
             nonzero_inputs.add(inp)
-            #zero_baseline, nonzero_baseline = calc_zero_and_nonzero_baselines(nonzero_inputs)
-            #print_baselines()
-        #print self.correlator.feng_eq_get().items()
+
+    def test_back2back_consistency(self):
+        """1. Check that back-to-back dumps with same input are equal"""
+        test_name = '{}.{}'.format(strclass(self.__class__), self._testMethodName)
+        init_dsim_sources(self.dhost)
+        test_chan=1500
+
+        requested_test_freqs = self.corr_freqs.calc_freq_samples(
+            test_chan, samples_per_chan=9, chans_around=1)
+        expected_fc = self.corr_freqs.chan_freqs[test_chan]
+        self.dhost.sine_sources.sin_0.set(frequency=expected_fc, scale=0.25)
+
+        for i, freq in enumerate(requested_test_freqs):
+            print ('Testing dump consistancy {}/{} @ {} MHz.'.format(
+                i+1, len(requested_test_freqs), freq/1e6))
+            self.dhost.sine_sources.sin_0.set(frequency=freq, scale=0.125)
+            dumps_data = []
+            for dump_no in range(3):
+                if dump_no == 0:
+                    this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                    init_data_dump = np.max(this_freq_dump['xeng_raw'])
+                else:
+                    this_freq_dump = self.receiver.data_queue.get(DUMP_TIMEOUT)
+                this_freq_data = this_freq_dump['xeng_raw']
+                dumps_data.append(this_freq_data)
+
+            diff_dumps = []
+            for comparison in range(1, len(dumps_data)):
+                d0 = dumps_data[comparison - 1]
```

Comment:
I would still like you to compare all subsequent dumps to the first one, just change to `d0 = dups_data[0]`